prboom2/src/r_bsp.c

```diff
@@ -375,6 +375,18 @@ static void R_AddLine (seg_t *line)
 
   if (V_IsOpenGLMode())
   {
+    line_t* l = line->linedef;
+    sector_t* sec = subsectors[currentsubsectornum].sector;
+
+    // Don't add plane to drawing list until we encounter a
+    // non-self-referencing linedef in the subsector.
+    if (sec->gl_validcount != validcount && (l == NULL || l->frontsector != l->backsector))
+    {
+      sec->gl_validcount = validcount;
```

Comment:
I'm a bit concerned about this decoupling. There are sector-related things in R_Subsector we want to only happen once, but that once is contingent on the conditions here, which may not be true. Doesn't that mean we may execute code from R_Subsector multiple times?